File: Stack/071.Simplify-Path/071.Simplify-Path.ts
function simplifyPath(path: string): string {
  const stack = [];
  const pathArray = path.split('/');
  for (let i = 0; i < pathArray.length; i++) {
    switch (pathArray[i]) {
      case '':
        // do nothing
        break;
      case '.':
        // do nothing
        break;
      case '..':
        stack.pop();
        break;
      default:
        stack.push(pathArray[i]);
    }
  }
  return `/${stack.join('/')}`;
}

File: Stack/071.Simplify-Path/071.Simplify-Path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        arr = path.split('/')
        stack = []
        for char in arr:
            if char == '' or char == '.':
                continue
            if char == '..':
                if len(stack) > 0:
                    stack.pop()
            else:
                stack.append(char)
        return '/' + '/'.join(stack)
                    
